executor/foreign_key.go

```diff
@@ -735,6 +824,8 @@ func genFKValueString(v types.Datum) (string, error) {
 	switch v.Kind() {
 	case types.KindInt64, types.KindUint64, types.KindFloat32, types.KindFloat64, types.KindMysqlDecimal:
 		return val, nil
+	case types.KindNull:
+		return "NULL", err
 	default:
 		return "'" + val + "'", nil
```

Comment:
Do all remaining types work well with `'`?

Reply:
nice catch, I just add all types test in `mysql_test/t/foreign_key_cascade_type.test` https://github.com/pingcap/tidb-test/pull/2000.